internal/fsutil/create_test.go

```diff
@@ -209,15 +279,25 @@ func (s *S) TestCreate(c *C) {
 
 		c.Assert(err, IsNil)
 		c.Assert(testutil.TreeDump(dir), DeepEquals, test.result)
+
 		// [fsutil.Create] does not return information about parent directories
 		// created implicitly. We only check for the requested path.
-		entry.Path = strings.TrimPrefix(entry.Path, dir)
-		// Add the slashes that TreeDump adds to the path.
-		slashPath := "/" + test.options.Path
-		if test.options.Mode.IsDir() {
-			slashPath = slashPath + "/"
+		if entry.Link != "" && entry.Mode&fs.ModeSymlink == 0 {
+			// Entry is a hard link.
+			pathInfo, err := os.Lstat(entry.Path)
+			c.Assert(err, IsNil)
+			linkInfo, err := os.Lstat(entry.Link)
+			c.Assert(err, IsNil)
+			os.SameFile(pathInfo, linkInfo)
```

Comment:
For hard links, we are testing if the created hard link is the same file as its link target. The reason we don't use `TreeDumpEntry` is because the `fsutil.Entry` for hard link contains neither the SHA256 nor the size, since creating hard links does not involve reading the data from the link target.

Reply:
Resolved, with a new comment added.